### PR TITLE
Feature/tao 10496/tao 10501/tao 10502 per interaction response processing

### PR DESCRIPTION
**Pull request by atsymuk**
Related to : 
- https://oat-sa.atlassian.net/browse/TAO-10496
- https://oat-sa.atlassian.net/browse/TAO-10501
- https://oat-sa.atlassian.net/browse/TAO-10502
 
- add configuration for per interaction response processing
- create interaction response outcome declarations when new interaction added to item
- rename interaction response outcome when response identifier changed
  
#### How to test
 
- prepare an instance of TAO
- create new test item 
- open authoring test item
- try to add some interaction
- edit response for this interaction
- check that response outcome declaration was added
- try to change response identifier
- check that outcome declaration was renamed
  
#### Dependencies
 
 - [ ] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/106

### CODE REVIEW

**Review by bziondik — COMMENTED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Comment by bziondik** [comment]
Unit tests fail
image

**Comment by atsymuk** [comment]
> Unit tests fail
> image


the test failing only locally, but on CI checks it is ok

File: config/default/qtiCreator.conf.php [hunk 1]
```diff
@@ -23,4 +23,4 @@
 /**
  * Default log config, does not lock resources
  */
-return ['multi-column' => false];
+return ['multi-column' => false, 'perInteractionRp' => false];
```
Comment (jsconan) [inline config/default/qtiCreator.conf.php:26]: A migration script would be needed to also take care of the update.

File: views/js/qtiCreator/model/mixin/editableInteraction.js [hunk 2]
```diff
@@ -85,6 +86,21 @@ define([
...
+                if(!interactionOutcome){
+                    interactionOutcome = item.createOutcomeDeclaration({
+                        cardinality : 'single',
+                        baseType : 'float'
+                    });
+
+                    interactionOutcome.attr('identifier', outcomeIdentifier);
+                }
+            }
+
```
Comment (jsconan) [inline views/js/qtiCreator/model/mixin/editableInteraction.js:103]: This could be moved to a dedicated helper since it used in several places

**Review by jsconan — COMMENTED** [review]
Do not forget to add the migration script to update the config.

Some improvements can be done on the code as the same pattern is used at several places.

And, yes, the `perInteractionRp` option could also be better managed. The loader should not be impacted.

**Review by bziondik — APPROVED** [review]

File: views/js/qtiCreator/itemCreator.js [hunk 3]
```diff
@@ -54,12 +54,13 @@ define([
...
      * @param {String} itemDataUrl - the data url
+     * @param {Boolean} perInteractionRp - per interaction processing enabled
      *
      * @returns {Promise} that resolve with the loaded item model
      */
-    var loadItem = function loadItem(uri, label, itemDataUrl){
+    var loadItem = function loadItem(uri, label, itemDataUrl, perInteractionRp){
         return new Promise(function(resolve, reject){
-            itemLoader.loadItem({uri : uri, label : label, itemDataUrl: itemDataUrl}, function(item){
+            itemLoader.loadItem({uri : uri, label : label, itemDataUrl: itemDataUrl, perInteractionRp }, function(item){
```
Comment (jsconan) [inline views/js/qtiCreator/itemCreator.js:63]:
```suggestion
            itemLoader.loadItem({uri, label, itemDataUrl, perInteractionRp }, function(item){
```

**Review by jsconan — APPROVED** [review]
Unit test is failing locally, but this is the same in develop.

The feature works well.

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by lecosson — APPROVED** [review]
Code review only
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**